### PR TITLE
[Plat-466] device id

### PR DESCRIPTION
**Pull request by chrisgray-vertex**
## Summary

Replacing the sessionId with a deviceId that gets stored and passed to the backend for keeping track of billing sessions.

## Test Plan

Load a scene and ensure that a device-id is generated and added to local storage. After rendering some frames, check the billing bucket for .json files that contain render frame events with the stored device-id.

### CODE REVIEW

File: packages/viewer/src/components/viewer/viewer.tsx [hunk 1]
```diff
@@ -333,7 +337,7 @@ export class Viewer {
    *
    * @private
    */
-  @Event() public sessionidchange!: EventEmitter<string>;
+  @Event() public deviceidchange!: EventEmitter<string>;
```
Comment (danschultz) [inline packages/viewer/src/components/viewer/viewer.tsx:340]: Let's use `deviceIdChange`, we're standardizing on camelcase event names vs the archaic DOM event types that are all lowercased.

File: packages/viewer/src/components/viewer/viewer.tsx [hunk 2]
```diff
@@ -1166,12 +1171,21 @@ export class Viewer {
...
+      this.deviceId = getStorageEntry(
+        StorageKeys.DEVICE_ID,
+        (entry) => entry['id']
       );
-    } else return this.sessionId;
+
+      if (this.deviceId == null) {
+        this.deviceId = UUID.create();
+        upsertStorageEntry(StorageKeys.DEVICE_ID, {
+          ['id']: this.deviceId,
```
Comment (danschultz) [inline packages/viewer/src/components/viewer/viewer.tsx:1184]: Let's name this key `device-id`.

File: packages/viewer/src/components/viewer/viewer.tsx [hunk 3]
```diff
@@ -1166,12 +1171,21 @@ export class Viewer {
...
 
-  private getSessionId(): string | undefined {
-    if (this.sessionId == null) {
-      return getStorageEntry(StorageKeys.STREAM_SESSION, (entry) =>
-        this.clientId ? entry[this.clientId] : undefined
+  private getDeviceId(): string | undefined {
+    if (this.deviceId == null) {
+      this.deviceId = getStorageEntry(
+        StorageKeys.DEVICE_ID,
+        (entry) => entry['id']
```
Comment (danschultz) [inline packages/viewer/src/components/viewer/viewer.tsx:1178]: ^

File: packages/viewer/src/components/viewer/viewer.spec.tsx [hunk 4]
```diff
@@ -318,26 +318,48 @@ describe('vertex-viewer', () => {
...
-            stream={stream}
-            sessionId={sessionId}
-          />
-        ),
+        template: () => <vertex-viewer clientId={clientId} stream={stream} />,
+      });
+
+      const storedDeviceId = Storage.getStorageEntry(
+        Storage.StorageKeys.DEVICE_ID,
+        (records) => records['id']
```
Comment (danschultz) [inline packages/viewer/src/components/viewer/viewer.spec.tsx:330]: ^

File: packages/viewer/src/components/viewer/viewer.tsx [hunk 5]
```diff
@@ -128,7 +132,7 @@ export class Viewer {
    *
    * @private
```
Comment (danschultz) [inline packages/viewer/src/components/viewer/viewer.tsx:133]: While you're in here, can you add `@internal` here so it doesn't show up in our Stencil docs?

File: packages/viewer/src/components/viewer/viewer.tsx (hunk 1 above)
Comment (chrisgray-vertex) [reply packages/viewer/src/components/viewer/viewer.tsx]: d33f010

File: packages/viewer/src/components/viewer/viewer.tsx (hunk 2 above)
Comment (chrisgray-vertex) [reply packages/viewer/src/components/viewer/viewer.tsx]: cdbde48

File: packages/viewer/src/components/viewer/viewer.tsx (hunk 3 above)
Comment (chrisgray-vertex) [reply packages/viewer/src/components/viewer/viewer.tsx]: cdbde48

File: packages/viewer/src/components/viewer/viewer.spec.tsx (hunk 4 above)
Comment (chrisgray-vertex) [reply packages/viewer/src/components/viewer/viewer.spec.tsx]: cdbde48

File: packages/viewer/src/components/viewer/viewer.tsx (hunk 5 above)
Comment (chrisgray-vertex) [reply packages/viewer/src/components/viewer/viewer.tsx:133]: 2e90a72

**Review by danschultz — APPROVED** [review]

**Pull request merged**